### ISSUE TITLE
nixos-install script 'mount: command not found'

### ISSUE THREAD

**Issue by aktaboot** [NixOS/nixpkgs#220211]
### Describe the bug
nixos-install script calls `mount` and `umount` but does not depend on them

On some systems, nixos-install will fail to find the mount and umount binaries on the system and therefore the install will get in a undesirable state. And the root password will not be prompted to the user.

@K900 suggested to use https://github.com/abathur/resholve 
@abathur 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"aarch64-linux"`
 - host os: `Linux 5.10.110-rockchip-rk3588, Debian GNU/Linux, 11 (bullseye), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/root/.nix-defexpr/channels/nixpkgs`
```

**Comment by abathur** [comment]
Ah. If we're talking about host system executables, I'm not sure resholve is going to be much help.

It'll complain about the missing commands and rewrite them to absolute paths if provided. We'd either have to prebuild for different systems or bring resholve to the system.

If the problem is just weird/inconsistent user paths, and if `mount` and `umount` are in ~normal locations, using `getconf PATH` to discover and set the ~normal system PATH may help?

**Comment by K900** [comment]
nixos-install is always installed with Nix, or at least should be, so we can (and should!) make it use Nix provided tools.

**Comment by abathur** [comment]
I don't really know, but I at least found two sources that make me think the specific uses of mount/umount here will work without a setuid wrapper: 
- https://github.com/NixOS/nixpkgs/pull/95444
- https://unix.stackexchange.com/questions/107500/why-does-mount-have-the-suid-bit-set

I decided to see how a naive attempt at using resholve would work but stopped when it was obvious that there's some dynamic behavior in the script (stringly-inlining other scripts). resholve won't be able to see through stuff like that without some changes to script source.

If you just want to get mount/umount in with minimal effort, I'd just add them to the existing makeBinPath. Note that `mount` also appears in nixos-enter.

If you're curious, I pushed up my ~starter on resholving it: (https://github.com/abathur/nixpkgs/commit/514afe19bea6ad2fa4f23270e5a5a1a0b19e2c4a)
